web/packages/teleterm/src/mainProcess/mainProcessClient.ts

```diff
@@ -265,3 +263,32 @@ export default function createMainProcessClient(): MainProcessClient {
     },
   };
 }
+
+/** Receives and confirms messages sent through `AwaitableSender`.*/
+function makeAwaitableReceiver<T>(listener: (value: T) => void): {
```

Comment:
This needs a better API and/or documentation. I assume you need to call `makeAwaitableReceiver` first and then somehow pass the resulting port to `AwaitableSender`? Sender and receiver are a pair so it'd be best if the whole pair was returned by a single function. But I suppose this is not possible because `AwaitableSender` needs to be created within the main process and cannot be passed from the renderer to the main process, but a port can be passed in such a way?

idk, I haven't reviewed the other PR yet but this seems hard to understand on its own.